includes/Handlers/Connection.php

```diff
@@ -272,20 +290,24 @@ public function handle_connect() {
 				throw new SV_WC_API_Exception( 'Invalid nonce' );
 			}
 
-			$merchant_access_token = ! empty( $_GET['merchant_access_token'] ) ? sanitize_text_field( $_GET['merchant_access_token'] ) : '';
+			$is_error                 = ! empty( $_GET['err'] ) ? true : false;
+			$error_code               = ! empty( $_GET['err_code'] ) ? sanitize_text_field( $_GET['err_code'] ) : '';
+			$merchant_access_token    = ! empty( $_GET['merchant_access_token'] ) ? sanitize_text_field( $_GET['merchant_access_token'] ) : '';
+			$system_user_access_token = ! empty( $_GET['system_user_access_token'] ) ? sanitize_text_field( $_GET['system_user_access_token'] ) : '';
+			$system_user_id           = ! empty( $_GET['system_user_id'] ) ? sanitize_text_field( $_GET['system_user_id'] ) : '';
+
+			if ( $is_error && $proxy_error_message = $this->get_message_for_error_code( $error_code ) ) {
+				throw new SV_WC_API_Exception( $proxy_error_message );
+			}
 
 			if ( ! $merchant_access_token ) {
 				throw new SV_WC_API_Exception( 'Access token is missing' );
 			}
 
-			$system_user_access_token = ! empty( $_GET['system_user_access_token'] ) ? sanitize_text_field( $_GET['system_user_access_token'] ) : '';
```

Comment:
I see this is an existing pattern mapping a missing field to a human-readable message, e.g. `! system_user_access_token => 'System User access token is missing' - and this is why you did the map of code to message above. I don't think that mapping is adding much value to the log messages - it's a layer of indirection, which is unhelpful IMO.

It would be useful to log the exact field that's missing for these errors, e.g. something like:

`Failed to connect to Facebook. No system_user_access_token provided.`

It looks like this code may have previously shown errors in the UI? I don't understand why the missing fields are converted to log messages like `System User access token is missing` - it's clearer to log the missing field / attribute IMO. 

Possible follow up - simplify and standardise this error handling, with a generic message (`Failed to connect to Facebook. No ${ field_name } provided.`. Or if we decide we still need the "translated" log messages as sentences, use a map/lookup table (similar to what you did for the new ones). I'm not clear on the implications here, so not a blocker, needs more discussion before follow up.